File: packages/java/parser-jvm-plugin-nonnull/src/test/java/dev/hilla/parser/plugins/nonnull/superclassmethods/CrudEndpoint.java
package dev.hilla.parser.plugins.nonnull.superclassmethods;

@EndpointExposed
public abstract class CrudEndpoint<T, ID> extends ReadOnlyEndpoint<T, ID> {
    public void delete(ID id) {
    }

    public T update(T entity) {
        return entity;
    }
}

File: packages/java/parser-jvm-plugin-nonnull-kotlin/src/test/kotlin/com/vaadin/hilla/parser/plugins/nonnull/kotlin/superclasses/endpointexposed/CrudEndpoint.kt
package com.vaadin.hilla.parser.plugins.nonnull.kotlin.superclasses.endpointexposed

import com.vaadin.hilla.parser.plugins.nonnull.kotlin.annotation.EndpointExposed

@EndpointExposed
abstract class CrudEndpoint<T, ID> : ReadOnlyEndpoint<T, ID>() {

    fun delete(id: ID) = Unit

    fun update(entity: T): T {
        return entity
    }

    fun save(entity: T?): T? {
        return entity
    }
}
